src/storage-client/src/client.rs

```diff
@@ -331,6 +331,29 @@ pub struct SinkStatisticsUpdate {
     pub bytes_committed: u64,
 }
 
+/// A source status update
+/// Represents a status update for a given object type. The inner value for each
+/// variant should be able to be packed into a status row that conforms to the schema
+/// for the object's status history relation.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct StatusUpdate<T = mz_repr::Timestamp> {
+    pub id: GlobalId,
+    pub status: String,
+    pub timestamp: T,
```

Comment:
this is because its part of `StorageResponse` :(

we only ever implement `ProtoType` for `Timestamp`, so idk why we still have that parameter on that